physics/eclipse_test.cpp

```diff
@@ -0,0 +1,167 @@
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+#include "physics/solar_system.hpp"
+#include "quantities/quantities.hpp"
+#include "quantities/si.hpp"
+#include "integrators/symplectic_runge_kutta_nyström_integrator.hpp"
+#include "geometry/epoch.hpp"
+#include "geometry/named_quantities.hpp"
+#include "quantities/elementary_functions.hpp"
+#include "geometry/grassmann.hpp"
+#include "testing_utilities/numerics.hpp"
+
+namespace principia {
+
+using astronomy::ICRFJ2000Equator;
+using geometry::JulianDate;
+using integrators::McLachlanAtela1992Order5Optimal;
+using quantities::si::Minute;
+using quantities::si::Metre;
+using quantities::si::Nano;
+using quantities::si::Milli;
+using quantities::si::Kilo;
+using quantities::ArcCos;
+using testing_utilities::AbsoluteError;
+using ::testing::Gt;
+using ::testing::Lt;
+using ::testing::AllOf;
+
+namespace physics {
+
+class EclipseTest : public testing::Test {
+ protected:
+  EclipseTest() {
+    solar_system_1950_.Initialize(
+        SOLUTION_DIR / "astronomy" / "gravity_model.proto.txt",
+        SOLUTION_DIR / "astronomy" /
+            "initial_state_jd_2433282_500000000.proto.txt");
+  }
+
+  SolarSystem<ICRFJ2000Equator> solar_system_1950_;
+};
+
+TEST_F(EclipseTest, Dummy) {
+  auto ephemeris = solar_system_1950_.MakeEphemeris(
+      McLachlanAtela1992Order5Optimal<Position<ICRFJ2000Equator>>(),
+      45 * Minute, 5 * Milli(Metre));
+
+  ephemeris->Prolong(JulianDate(2433374.5));  // Prolong until date of eclipse
+                                              // (Eclipse was 1950-04-02 but JD
+                                              // is 1950-04-03:00:00:00)
+  // pass body to Ephemeris.trajectory
+  auto sun = ephemeris->bodies()[solar_system_1950_.index("Sun")];
+  auto earth = ephemeris->bodies()[solar_system_1950_.index("Earth")];
+  auto moon = ephemeris->bodies()[solar_system_1950_.index("Moon")];
+
+  // Massive_body eventually needs radius information. Or non-hardcoded data
```

Comment:
`Massive_body` should be spelled `MassiveBody`.

More importantly, could you please open an issue for adding the radius to `MassiveBody`?  (And then you can reference the issue in this comment.)